src/pages/atomicdex/api/v20/index.mdx

```diff
@@ -101,3 +101,647 @@ It includes a uniform request, successful and error response formats. At the mom
+| coin            | string         | The coin which will be used to pay the transaction fee.                           |
+| gas             | integer        | `Eth` type only. The amount of gas to be used for the transaction.                |
+| gas\_price      | numeric string | `Eth` or `Qrc20` type only. Price per unit of gas to be used for the transaction. |
+| gas\_limit      | numeric string | `Tendermint` or `Qrc20` type only. Maximum gas to be used for the transaction.    |
+| miner\_fee      | numeric string | `Tendermint` type only. Fee to mine the transaction.                              |
+| total\_fee      | numeric string | `Eth` type only. Gas price multiplied by gas amount.                              |
+| total\_gas\_fee | numeric string | `Qrc20` type only. Gas price multiplied by gas amount.                            |
+
+<CollapsibleSection expandedText="Hide Examples" collapsedText="Show Examples">
+  #### Example of Eth type
+
+  ```json
+  {
+    "type": "Eth",
+    "coin": "BNB",
+    "gas": 109739,
+    "gas_price": "0.000000003",
+    "total_fee": "0.000329217"
+  }
+  ```
+
+  #### Example of Qrc20 type
+
+  ```json
+  {
+    "type": "Qrc20",
+    "coin": "tQTUM",
+    "miner_fee": "0.00000447",
+    "gas_limit": 100000,
+    "gas_price": 40,
+    "total_gas_fee": "0.04"
+  }
+  ```
+
+  #### Example of Tendermint type
+
+  ```json
+  {
+    "type": "Tendermint",
+    "coin": "IRIS",
+    "amount": "0.038553",
+    "gas_limit": 100000
+  }
+  ```
+
+  #### Example of Utxo type
+
+  ```json
+  {
+    "type": "Utxo",
+    "amount": "0.00001"
+  }
+  ```
+</CollapsibleSection>
+
+### WithdrawNftData
+
+The `WithdrawNftData` object is used for withdrawals of NFTs on ERC721 and ERC1155 contracts. It includes the following items for a given coin or token:
+
+| Parameter      | Type    | Description                                                                                                                              |
+| -------------- | ------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
+| chain          | string  | The token chain. Chain must be [activated](/atomicdex/api/legacy/coin_activation/) first.                                                |
+| to             | string  | Destination address to withdraw the token to.                                                                                            |
+| token\_address | string  | Token address.                                                                                                                           |
+| token\_id      | string  | Token ID.                                                                                                                                |
+| fee            | object  | A standard [WithdrawFee](/atomicdex/api/v20/#withdraw-fee) object. May be missing for older transfers.                                   |
```

Comment:
Seems like `[WithdrawFee](/atomicdex/api/v20/#withdraw-fee)` is an object, which we have in withdraw fungible token **response**. 

In nft withdraw request I used `fee: Option<WithdrawFee>`(rust version), which is used in coins withdraw **requests**.
```rust
#[serde(tag = "type")]
pub enum WithdrawFee {
    UtxoFixed {
        amount: BigDecimal,
    },
    UtxoPerKbyte {
        amount: BigDecimal,
    },
    EthGas {
        /// in gwei
        gas_price: BigDecimal,
        gas: u64,
    },
    Qrc20Gas {
        /// in satoshi
        gas_limit: u64,
        gas_price: u64,
    },
    CosmosGas {
        gas_limit: u64,
        gas_price: f64,
    },
}
```
Looks like this [object](https://github.com/KomodoPlatform/komodo-docs-mdx/blob/add-nft-docs/src/pages/atomicdex/api/common_structures/index.mdx#L150) . Actually there is a repetition in another [place](https://github.com/KomodoPlatform/komodo-docs-mdx/blob/add-nft-docs/src/pages/atomicdex/api/v20/index.mdx#L381). I think we should leave one example in `common_structures` folder.


`The FeeInfo response object includes the following items for [withdraw (v2)](/atomicdex/api/v20/withdraw/) requests`

`WithdrawFee` enum, which is `FeeInfo` in docs, is for requests, not response
```rust
pub struct WithdrawRequest {
    coin: String,
    from: Option<WithdrawFrom>,
    to: String,
    #[serde(default)]
    amount: BigDecimal,
    #[serde(default)]
    max: bool,
    fee: Option<WithdrawFee>,
    memo: Option<String>,
    /// Currently, this flag is used by ETH/ERC20 coins activated with MetaMask **only**.
    #[cfg(target_arch = "wasm32")]
    #[serde(default)]
    broadcast: bool,
}
```


and this `TxFeeDetails` enum we have in responses, it is `WithdrawFee` in your docs.
```rust
/// Please note that no type should have the same structure as another type,
/// because this enum has the `untagged` deserialization.
#[derive(Clone, Debug, PartialEq, Serialize)]
#[serde(tag = "type")]
pub enum TxFeeDetails {
    Utxo(UtxoFeeDetails),
    Eth(EthTxFeeDetails),
    Qrc20(Qrc20FeeDetails),
    Slp(SlpFeeDetails),
    Tendermint(TendermintFeeDetails),
    #[cfg(all(
        feature = "enable-solana",
        not(target_os = "ios"),
        not(target_os = "android"),
        not(target_arch = "wasm32")
    ))]
    Solana(SolanaFeeDetails),
}
```

Reply:
updated. thanks, there were a half a dozen or so other duplicated structures. I've removed all that was in the `dev` and `v2-dev` folders and added them where appropriate under the `common_structures` folder. The `FeeInfo` object used for responses and the `WithdrawFee` object used for requests shoudl now be properly applied within the NFT docs.

Reply:
@smk762 

`FeeInfo` should be for requests, 
https://github.com/KomodoPlatform/komodo-docs-mdx/blob/418d2d443d11271605bab424d4a6cc30313bfe89/data-for-gpts/komodefi-api/all-api-content.txt#L341

Here in `WithdrawNftData` in docs we have `WithdrawFee`, but in rust repo `WithdrawNftData` is for reqs. `FeeInfo` should be her.
https://github.com/KomodoPlatform/komodo-docs-mdx/blob/418d2d443d11271605bab424d4a6cc30313bfe89/data-for-gpts/komodefi-api/v20-api-content.txt#L3321-L3331

and `WithdrawFee` should be for response.
https://github.com/KomodoPlatform/komodo-docs-mdx/blob/418d2d443d11271605bab424d4a6cc30313bfe89/data-for-gpts/komodefi-api/v20-api-content.txt#L3256-L3270
Thus `| gas             | integer        | `Eth` type only. The amount of gas to be used for the transaction.` not to be used, but gas was used

Just in rust code we have another names.
rust `WithdrawFee` enum is for req
```rust
#[derive(Clone, Debug, Deserialize, PartialEq)]
#[serde(tag = "type")]
pub enum WithdrawFee {
    UtxoFixed {
        amount: BigDecimal,
    },
    UtxoPerKbyte {
        amount: BigDecimal,
    },
    EthGas {
        /// in gwei
        gas_price: BigDecimal,
        gas: u64,
    },
    Qrc20Gas {
        /// in satoshi
        gas_limit: u64,
        gas_price: u64,
    },
    CosmosGas {
        gas_limit: u64,
        gas_price: f64,
    },
}
```

This is for responses
```rust
#[derive(Clone, Debug, PartialEq, Serialize)]
#[serde(tag = "type")]
pub enum TxFeeDetails {
    Utxo(UtxoFeeDetails),
    Eth(EthTxFeeDetails),
    Qrc20(Qrc20FeeDetails),
    Slp(SlpFeeDetails),
    Tendermint(TendermintFeeDetails),
    #[cfg(all(
        feature = "enable-solana",
        not(target_os = "ios"),
        not(target_os = "android"),
        not(target_arch = "wasm32")
    ))]
    Solana(SolanaFeeDetails),
}


// for example eth tx fee
pub struct EthTxFeeDetails {
    pub coin: String,
    pub gas: u64,
    /// WEI units per 1 gas
    pub gas_price: BigDecimal,
    pub total_fee: BigDecimal,
}
```

I think you should rename `WithdrawFee` type in docs to `TxFeeDetails`, and `FeeInfo` rename to `WithdrawFee`. So it will be easier for all of us, as we will have same object names in both rust code and docsx. Se you and me can check what fee object should be in docs according rust code name.